src/Cake.NuGet/Install/NuGetPackageInstaller.cs

```diff
@@ -95,38 +96,36 @@ public IReadOnlyCollection<IFile> Install(PackageReference package, PackageType
                 return Array.Empty<IFile>();
             }
 
-            if (!_projects.TryGetValue(type, out var project))
-            {
-                var pathResolver = new PackagePathResolver(packageRoot);
-                project = new NugetFolderProject(_fileSystem, _contentResolver, _config, _log, pathResolver, packageRoot, targetFramework);
-                _projects.Add(type, project);
-
-                if (!project.Root.Equals(packageRoot))
-                {
-                    // This should not happen since all addins/tools are installed to same directory.
-                    throw new ArgumentException($"Path is not same as previous package of type: {type}", nameof(path));
-                }
-            }
+            var pathResolver = new PackagePathResolver(packageRoot);
+            var project = new NugetFolderProject(_fileSystem, _contentResolver, _config, _log, pathResolver, packageRoot, targetFramework);
             var packageManager = new NuGetPackageManager(sourceRepositoryProvider, _nugetSettings, project.Root)
             {
                 PackagesFolderNuGetProject = project
             };
-
             var sourceRepositories = sourceRepositoryProvider.GetRepositories();
-            var includePrerelease = false;
-            if (package.Parameters.ContainsKey("prerelease"))
-            {
-                bool.TryParse(package.Parameters["prerelease"].FirstOrDefault() ?? bool.TrueString, out includePrerelease);
-            }
-            var resolutionContext = new ResolutionContext(DependencyBehavior.Lowest, includePrerelease, false, VersionConstraints.None, project.GatherCache);
+            var includePrerelease = package.IsPrerelease();
+            var dependencyBehavior = GetDependencyBehavior(type, package);
+            var resolutionContext = new ResolutionContext(dependencyBehavior, includePrerelease, false, VersionConstraints.None, _gatherCache);
             var projectContext = new NuGetProjectContext(_log);
+
             packageManager.InstallPackageAsync(project, packageIdentity, resolutionContext, projectContext,
                 sourceRepositories, Array.Empty<SourceRepository>(),
                 CancellationToken.None).Wait();
 
             return project.GetFiles(path, package, type);
         }
 
+        private DependencyBehavior GetDependencyBehavior(PackageType type, PackageReference package)
+        {
+            if (type == PackageType.Addin)
+            {
+                return package.ShouldLoadDependencies(_config) ?
```

Comment:
You could change this to

```csharp
if(package.ShouldLoadDependencies(_config))
{
    return DependencyBehavior.Lowest;
}
```

since the default behavior is to ignore dependencies by exiting the scope. Just a personal preference of course. Do as you wish 😄